lib/breadcrumbs_on_rails/json_ld/builder.rb

```diff
@@ -19,11 +20,32 @@ def render_element(element, index)
           "@type" => "ListItem",
           "position" => index,
           "item" => {
-            "@id" => "#{@context.root_url.chop}#{compute_path(element)}",
+            "@id" => compute_url(element),
             "name" => compute_name(element),
           },
         }
       end
+
+      def compute_url(element)
+        root_url_with_default_options = URI.parse(@context.root_url)
+
+        # Get rid of the duplicate path and query parameters;
+        # since `default_url_options` affects every helper method dependent on
+        # `url_for` (e.g. root_path, root_url), both `url_for`, internally used
+        # in `compute_name` provided by Breadcrumbs On Rails, and `root_url`
+        # may have the same information set by `default_url_options`.
+        root_url =
+          URI.scheme_list[root_url_with_default_options.scheme.upcase].build(
+            host: root_url_with_default_options.host,
+            userinfo: root_url_with_default_options.userinfo,
+            port: root_url_with_default_options.port,
+          )
+
+        # The returned value of `compute_path(element)` may contain the default
+        # options as path or query parameters. This is why this value shouldn't
+        # be passed in `URI::Generic.build` as a `path` option directly.
+        URI.join(root_url, compute_path(element))
```

Comment:
I'm not sure we should return a URI object or a string object, but it works.
I believe `JSON.pretty_generate` internally calls `to_s` to each object.